File: Lecture07/DynamicFragment/app/src/main/java/com/codingblocks/dynamicfragment/MainActivity.java
package com.codingblocks.dynamicfragment;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnAdd = findViewById(R.id.btnAdd);

        final FragmentManager fragmentManager = getSupportFragmentManager();

//        MyFragment myFragment = (MyFragment) fragmentManager.findFragmentByTag("FIRST_INSTANCE");

        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Add a new Fragment dynamically

                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

//                fragmentTransaction.add(new MyFragment(), "FIRST_INSTANCE");

//                fragmentTransaction.add(R.id.container,new MyFragment());

//                fragmentTransaction.add(R.id.container,new MyFragment(),"A_UNIQUE_TAG");

                fragmentTransaction.replace(R.id.container, new MyFragment(), "UNIQUE_TAG");

                fragmentTransaction.commit();
            }
        });

    }
}

File: Lecture14/NotesDb/app/src/main/java/com/codingblocks/notesdb/MainActivity.kt
package com.codingblocks.notesdb

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val noteDb by lazy {
        NoteDatabase(this)
    }

    private val notes by lazy {
        noteDb.getAllNotes()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val noteAdapter = NotesAdapter(notes)

        rvNotes.layoutManager = LinearLayoutManager(this)

        rvNotes.adapter = noteAdapter

        btnSubmit.setOnClickListener {

            val note = Note(etTitle.text.toString(), etDesc.text.toString())

            //Save this note to the db
            val id = noteDb.insertNote(note)

            if (id != -1L) {

                val insertedNote: Note? = noteDb.getNoteWithId(id)

                insertedNote?.let { fetchedNote ->
                    notes.add(fetchedNote)
                    noteAdapter.notifyItemInserted(notes.size - 1)
                }

            } else
                Toast.makeText(this, "There was an error saving this to the db", Toast.LENGTH_SHORT).show()

        }

    }
}
